src/TCommandLine.cpp

```diff
@@ -693,6 +703,69 @@ void TCommandLine::mousePressEvent(QMouseEvent* event)
             QAction* action_addWord = nullptr;
             QAction* action_removeWord = nullptr;
             QAction* action_dictionarySeparatorLine = nullptr;
+
+            QActionGroup* actionGroup_commandHistory = new QActionGroup(popup);
```

Comment:
This adds 3 lines plus a separator to the context menu - but the spelling can add ~~loads and loads and  loads~~ ***lots*** more. If you are typing an ultra secret password and want to be certain it isn't there do you want to load in an additional Lua module that might do something sneaky instead (but makes the context menu 1 more line plus a separator normally or 2 lines plus 2 separators when the module is used) or  just right click on the context menu and select an option that is known to be hard coded into the application and always visible there and becomes familiar with repeated use.

In my mind your UI/UX does not sound *that* compelling...

Reply:
> but the spelling can add loads lots more. 

Yeah, and having tried it I think it is too verbose (adds 5 lines even when there's nothing). I don't want to go down this route here too. Way too few people will have a need for this, in too little scenarios, so how much UI space it'll take up is not in balance.

Reply:
Ah,  I've just thought of something about the UI I did that wouldn't fly so well with splitting the functionality up into two separate things:

When you select the "forget the next line only" option it stays in that state until a command is entered into that command line (even between sessions) *then* once a command has been entered it jumps back to the "Save command history" option - so it gives a visible (post-usage) indication that it has taken place.

If the "don't remember option" is selected then "forget the next line" is largely pointless (not entirely because if it is selected from either of the other states then any instances of the command then given is nuked from the current session command history - and if that history *is* saved, from that as well).  By doing this whilst "disconnected" it also provides a means to remove a command (say an entered password) from the history file should one have accidentally have gotten saved without having to track down the history file and edit it by hand - which is possible BTW.

Anyhow, I don't think it would be easy to separate out the two different features and leave one in the context menu and the other down to a Lua package and provide the (relatively simple):
* "save command history"
* "don't save the next command" - changing back to "save command history" after next command
* "don't save command history"

UI.